### ISSUE TITLE
Make elections exclusive: disallow candidates from running in multiple elections

### ISSUE THREAD

**Issue by BjarniRunar** [piratar/wasa2il#187]
It would be useful to be able to prevent users from running in multiple elections at the same time. This would be an optional per-election setting.

This is a moderatly complicated SQL query, but should be doable: find all the elections that overlap in time with this one, if the candidate is running in any of them, prevent them from running in this one.

**Comment by helgihg** [comment]
I'm wondering; is this really needed? While it's simple in principle, it still begs questions like whether people should be excluded from elections in other polities. Have we ever had the problem of people running in two different elections that they should not have been allowed to run in?

**Comment by BjarniRunar** [comment]
I think this was a request from one of the campaign mgmt teams, I don't know if it ever came up for real. I forget, sorry!

You're a popular guy - if you were to run in both Reykjavik North and Reykjavik South (assuming they were separate) you would be likely to end up on both ballots, which would then need to be "fixed" by hand after the primary, invalidating a bunch of people's votes. Making candidates declare up front where they are actually running avoids wasted votes. So it's a good idea from that POV.

But whether it needs to be automated is another matter. It could obviously be handled relatively easily from the human side. So I'd say it's not really needed, but it would be nice to have since it would prevent arguing and stress (which seem to be major features of volunteer-run primaries).

**Comment by helgihg** [comment]
I can certainly see the problem from a theoretical standpoint, but it kind of stops there. These online elections are always held over the course of at least a week, so spotting a double candidate (hey, that's a cool term) when they're running, or even when the election has already begun, seems like a much easier solution.

I just wonder if this is really worth the effort, because while it's simple in principle, it somehow always spirals out of control in my head, when I start taking polities into account and different rules in different elections.

So, let's say election A in Bulltown is exclusive and candidate Zaz runs. Election B in Beartown is not exclusive, and Zaz decides to run there as well. Can she? Is she prevented from running in B, even though B has no problem with double candidates, or does she automatically pop out of A? To complicate things, what if there's another election C in Bulltown as well, which is not exclusive toward A? - I somehow always end up thinking that it's easy to do this badly but difficult to do this well, and this is a feature that would need to be done well to avoid frantic calls when voting is about to start and something doesn't work the way people expect it to. Even just configuring this properly seems overly complicated from a user's point of view.

This leads me to believe that it's easier to handle this manually during the collecting of candidates or even during voting. The latter would depend on issue #120 being completed.

What do you say we close this issue but instead implement #120 so that officers or admins or whatever can kick candidates out of elections before the votes are counted?

**Comment by BjarniRunar** [comment]
Sounds reasonable to me!

I think all that's necessary for this to be easy to handle manually, without requiring a full implementation of #120, is for people running elections to take care to have a gap between "deadline for announcing candidacy" and "beginning of elections."

If there is always a reasonable gap (a few hours) between those two deadlines, the people running the elections will have time to implement whatever due diligence is required before voting begins.

I don't remember if the current system allows for such a gap, it's been a while since I looked at the election settings page. I think it does?

**Comment by helgihg** [comment]
Yep, it sure does!

Okay, I'll just close this issue and then we're immediately a bit closer to getting to #120. ;)

In fact, I'll just put that next(-ish) on my plate to brutally slaughter two birds with one blunt murder weapon.
